backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/comparatorsource/impl/LuceneTextFieldComparatorSource.java

```diff
@@ -50,7 +51,8 @@ else if ( SortMissingValue.MISSING_LAST.equals( missingValue ) ) {
 		TextMultiValuesToSingleValuesSource source =
 				TextMultiValuesToSingleValuesSource.fromField( fieldname, multiValueMode, nestedDocsProvider );
 
-		return new FieldComparator.TermOrdValComparator( numHits, fieldname, considerMissingHighest ) {
+		// forcing to not skipping documents
+		return new TermOrdValComparator( numHits, fieldname, considerMissingHighest, reversed, false ) {
```

Comment:
I'm confused, why do we need this? Do we get a wrong value in `enableSkipping`? If so, shouldn't we rather investigate why we're getting a wrong value?

Reply:
`enableSkipping` originally comes from the `FieldValueHitQueue`:
```java
for (int i = 0; i < numComparators; ++i) {
      SortField field = fields[i];
     // ... 
     // enableSkipping = i == 0 
      comparators[i] = field.getComparator(size, i == 0);
    }
```
So there's not much we can do about this, unless we pass a dummy field into sorting, which I don't think we want to do...
Then the value can be set to false afterwards by calling one of `SortField#setOptimizeSortWithIndexedData()` `SortField#setOptimizeSortWithPoints()` but both are deprecated for removal in Lucene 10.

Text sort works ok for as long as we don't do a multi-index search where one of the indexes is missing a field... 
I guess we could check if it's a multiindex search in the `AbstractLuceneDocumentValueSort` constructor and if so set `sortField.setOptimizeSortWithIndexedData( false );` but that method is already deprecated 🙈 

I've also looked a bit closer at Geopoint sorting... numeric comparators are more or less the same between two versions. Problem is that we are getting a `new IllegalArgumentException( "Field " + field + " is indexed with multiple dimensions, sorting is not supported");` 
From what it looks in 8.11:
```java
  // For numeric sort fields, if true, indicates the DocValues and Point fields
  // with the same name have the exactly same data indexed.
  // This allows to use them to optimize sort and skip non-competitive documents.
  private boolean canUsePoints = false;
```
This flag that enables optimization is off. And it is never flipped (we never call `SortField#setCanUsePoints()`).
But in 9.5 similar flag:
```java
  // Indicates if sort should be optimized with indexed data. Set to true by default.
  @Deprecated private boolean optimizeSortWithIndexedData = true;
```
is enabled by default. Which leads to:
```java
if (info.getPointDimensionCount() > 1) {
          throw new IllegalArgumentException(
              "Field " + field + " is indexed with multiple dimensions, sorting is not supported");
        }
```

And since you've mentioned Elasticsearch in the other place -- I've looked at what they are doing for their comparators and ... ```return new LongComparator(numHits, null, null, reversed, enableSkiping: false) {``` 🥲

Reply:
Ok, ok, I'll trust you. I guess this boolean is just a hint. Thanks for looking into this.

Someday, we might want to investigate whether we can make skipping actually work in at least some cases, but that will probably require support for index-time sorting (where documents are stored in the index already sorted), so it's definitely low priority.